File: src/main/webapp/JS/estoque.js
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

const estoque = Array.from(document.getElementsByClassName("stock-card"));

const showStockModal = () => {
    document.getElementById("overlay").style = "display: flex";
};

const closeStockModal = () => {
    document.getElementById("overlay").style = "display: none";
};

const addProductToStock = (idProduto, idUnidade, quantidade) => {
    if (idUnidade && idProduto) {
        const uri = generateURI(idProduto, idUnidade, quantidade);

        axios.get(`./CadastroEstoqueServlet${uri}`)
                .then((response) => console.log(response))
                .catch((error) => console.log(error))
                .then(closeStockModal());
    } else {
        closeStockModal();
    }

};

const updateProductStock = (id, idProduto, idUnidade, quantidade) => {
    const uri = generateURI(idProduto, idUnidade, quantidade, id);

    axios.get(`./EditarEstoqueServlet${uri}`)
            .then((response) => console.log(response))
            .catch((error) => console.log(error))
            .then(closeStockModal());

};

const generateURI = (idProduto, idUnidade, quantidade, id = null) => {
    let uri;

    if (id) {
        uri = `?id=${id}&idProduto=${idProduto}&idUnidade=${idUnidade}&quantidade=${quantidade}`;
    } else {
        uri = `?idProduto=${idProduto}&idUnidade=${idUnidade}&quantidade=${quantidade}`;
    }
    return uri;
};

const unityFilter = (unityId) => {
    const cardGalery = document.getElementById("card-galery");
    cardGalery.innerHTML = "";

    if (unityId) {
        const filteredList = estoque.filter((element) => element.getAttribute("uid") == unityId);
        filteredList.forEach((element) => cardGalery.appendChild(element));

    } else {
        estoque.forEach((element) => cardGalery.appendChild(element));
    }
};

const submitStockUpdate = (idProduto, idUnidade, quantidade) => {
    const filter = estoque.filter((element) => element.getAttribute("pid") == idProduto && element.getAttribute("uid") == idUnidade);

    if (filter.length) {
        const id = filter[0].getAttribute("key");
        updateProductStock(id, idProduto, idUnidade, quantidade);
    } else {
        addProductToStock(idProduto, idUnidade, quantidade);
    }

    window.location.reload();
    return false;
};

File: src/main/java/model/Estoque.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

/**
 *
 * @author Marcelo
 */
public class Estoque {
    private int id;
    private int idProduto;
    private int idUnidade;
    private int quantidade;

    public Estoque(int idProduto, int idUnidade, int quantidade) {
        this.idProduto = idProduto;
        this.idUnidade = idUnidade;
        this.quantidade = quantidade;
    }

    public Estoque(int id, int idProduto, int idUnidade, int quantidade) {
        this.id = id;
        this.idProduto = idProduto;
        this.idUnidade = idUnidade;
        this.quantidade = quantidade;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getIdProduto() {
        return idProduto;
    }

    public void setIdProduto(int idProduto) {
        this.idProduto = idProduto;
    }

    public int getIdUnidade() {
        return idUnidade;
    }

    public void setIdUnidade(int idUnidade) {
        this.idUnidade = idUnidade;
    }

    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }
    
    
}
